shared/api-definitions-ex.asciidoc

```diff
@@ -0,0 +1,30 @@
+[[sample-object]]
+=== Object1 API objects
+++++
+<titleabbrev>Object1</titleabbrev>
+++++
+
+//Provide a brief description
+
+Is a cool thing.
+
+// Guidelines for API object documentation
+// ***************************************
+// * Use a definition list.
+// * Each property should be marked as Optional or Required.
+// * Include the data type.
+// * Include default values as the last sentence of the first paragraph.
+// * Include a range of valid values, if applicable.
+// * For nested objects, link to a separate definition list.
+// ***************************************
+
+[float]
+==== Properties
+
+////
+For example:
+
+`analysis_config`::
+  (object) The analysis configuration, which specifies how to analyze the data.
+  See <<ml-analysisconfig, analysis configuration objects>>.
+////
```

Comment:
Let's include an example, similar to how it's handled in the Cloud Docs: https://www.elastic.co/guide/en/cloud-enterprise/current/definitions.html#AllocatedInstancePlansInfo

```suggestion
////

==== Example
---
{
   "pending" : true,
   "version" : "string",
   "zone_count" : 0
}
---
// NOTCONSOLE
```

Reply:
I think if we add examples like this, they'll get out-dated quickly unless we add ways to test them. That's the content that gets out-dated fastest on the APIs I've worked on.

Reply:
Good point. I'm okay with leaving them out for now. I can't think of any way to make these testable without a lot of overhead.